src/components/ItemList/ItemList.tsx

```diff
@@ -18,26 +18,45 @@ const ItemList = <T extends { id: string }>({
     onActivate = noop,
     onSelect,
     ...rest
-}: Props<T>): JSX.Element => (
-    <ul className="ba-ItemList" data-testid="ba-ItemList" role="listbox" {...rest}>
-        {items.map((item, index) => (
-            <ItemRowComponent
-                key={item.id}
-                className="ba-ItemList-row"
-                isActive={index === activeItemIndex}
-                item={item}
-                onClick={(event: React.SyntheticEvent) => {
-                    onSelect(index, event);
-                }}
-                onMouseDown={(event: React.SyntheticEvent) => {
-                    event.preventDefault(); // preventDefault on mousedown so blur doesn't happen before click
-                }}
-                onMouseEnter={() => {
-                    onActivate(index);
-                }}
-            />
-        ))}
-    </ul>
-);
+}: Props<T>): JSX.Element => {
+    const activeRef = React.useRef<ItemRowRef>(null);
+    const listRef = React.useRef<HTMLUListElement>(null);
+
+    React.useEffect(() => {
+        const { current: listEl } = listRef;
+        const { current: activeEl } = activeRef;
+
+        if (listEl && activeEl) {
+            if (listEl.scrollTop > activeEl.offsetTop) {
```

Comment:
Can we cache these values to avoid calling them repeatedly? The fewer interactions with the DOM, the better.

Reply:
Put a different way, can we use destructuring for these properties?